<?xml version="1.0"?>
<launch>
  <!--node pkg="nodelet" type="nodelet" name="pcl_manager"  args="manager" output="screen"/-->

  <arg name="start_manager" default="true"/>
  <arg name="nodelet_manager" default="extract_planes_manager"/>
  <node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>

   <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
  <!--node pkg="nodelet" type="nodelet" name="voxel_grid2" args="load pcl/VoxelGrid cam3d_nodelet_manager" output="screen">
    <remap from="~input" to="/aggregate_point_map/pc_aligned"/>
    <rosparam>
      filter_field_name: z
      filter_limit_min: 0.2
      filter_limit_max: 3
      filter_limit_negative: False
      leaf_size: 0.02
      input_frame: "/map"
      output_frame: "/map"
    </rosparam>
  </node-->

  <node pkg="nodelet" type="nodelet" name="plane_extraction" args="load cob_3d_segmentation/PlaneExtractionNodelet $(arg nodelet_manager)" output="screen">
    <remap from="point_cloud2" to="registration/pc_aligned"/>
    <!-- remap from="point_cloud2" to="/cam3d/rgb/points"/-->
    <remap from="chull" to="plane_extraction/chull"/>
    <remap from="visualization_marker" to="plane_extraction/visualization_marker"/>
    <remap from="object_cluster" to="plane_extraction/object_cluster"/>
    <remap from="polygons" to="plane_extraction/polygons"/>
    <remap from="polygon_array" to="plane_extraction/polygon_array"/>
    <remap from="shape_array" to="segmentation/shape_array"/>
    <!--remap from="point_cloud2" to="/aggregate_point_map/map"/-->
    <!--remap from="point_cloud2" to="/voxel_grid2/output"/-->
    <param name="file_path" type="string" value="/tmp"/>
    <param name="save_to_file" type="bool" value="false"/>
    <param name="plane_constraint" type="int" value="0"/>

    <!-- Voxel filter parameters -->
    <param name="vox_leaf_size" type="double" value="0.04"/>
    <param name="passthrough_min_z" type="double" value="-0.1"/>
    <param name="passthrough_max_z" type="double" value="3.0"/>

    <!-- Clustering parameters -->
      <param name="cluster_tolerance_" type="double" value="0.06"/>
    <param name="min_cluster_size_" type="int" value="50"/>

    <!-- Normal Estimation parameters -->
      <param name="radius" type="double" value="0.1"/>
    <!-- Concave Hull parameters -->
      <param name="alpha" type="double" value="0.05"/>

      <!-- Segmentation parameters -->
    <!-- param name="normal_distance_weight" type="double" value="0.5"/>
    <param name="max_iterations" type="int" value="100"/>
    <param name="distance_threshold" type="double" value="0.03"/-->
  </node>

</launch>
